File: Lab3/eshop-ui/src/app/models/Product.ts
import {ProductComment} from './ProductComment';
import {ObjectBase} from './ObjectBase';

export class Product extends ObjectBase {
  categoryId: string;
  imageUrl: string;
  price: number;
  ratingSum: number;
  ratingCount: number;
  raters: string[];
  productComments: ProductComment[];
}

File: Lab4/eshop-app/src/main/java/com/yube/model/dto/Product.java
package com.yube.model.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.util.List;

@Data
@ToString(exclude = {"productComments"}, callSuper = true)
@EqualsAndHashCode(exclude = {"productComments"}, callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class Product extends ObjectBase {
    private String categoryId;
    private String imageUrl;
    private Double price;
    private Integer ratingSum;
    private Integer ratingCount;
    private List<String> raters;
    private List<ProductComment> productComments;
}
